### ISSUE TITLE
problem with TIMESTAMPDIFF postgresql implementation 

### ISSUE THREAD

**Issue by josensanchez** [oroinc/doctrine-extensions#34]
giving 2 Dates: 2017-02-16 and 2016-10-01 : 

the diff is 
0 years 4 mons 15 days

```
TIMESTAMPDIFF('year', '2016-10-01', '2017-02-16')  // MySql returns:0 PostgreSql returns:1
TIMESTAMPDIFF('month', '2016-10-01', '2017-02-16') //MySql returns:4 PostgreSql returns:4
```

This is a problem if someone is trying to calculate people's age. So I changed the year implementation

```
    protected function getDiffForYear(Node $firstDateNode, Node $secondDateNode, SqlWalker $sqlWalker)
    {

        return sprintf(
            'EXTRACT(YEAR FROM (age(%s, %s)))',
            $this->getExpressionValue($secondDateNode, $sqlWalker),
            $this->getExpressionValue($firstDateNode, $sqlWalker)
        );
    }
```

But then the months calculation began to return negative values. So I changed it too... but I did not find an elegant way to solve it.
The thing is that when you get an interval you do not even know witch months are you dealing with

Using the number of days / 30 is close but not quite right

and I'm not even sure how to began to solve the thing with quarters...

I ended up leaving everything as it was, but I changed the name  of the function getDiffForYear() to getDiffForYearAux() and only use it for calculate months and quarters as it was before and for years I use my function getDiffForYear()

Maybe someone can find a better solution

**Comment by x86demon** [comment]
Hi, @josensanchez .
Thank you for posting detailed issue explanation. It is known issue, as there are no equivalent functions in PostgreSQL that may work same to MySQL`s  TIMESTAMPDIFF. The only solution is to backport logic implementation of this function from MySQL to SQL variant and return it for PostgreSQL platform. But it will be to heavy and slow. If possible - avoid TIMESTAMPDIFF usage. Current implementation is based on threads HOW TO implement TIMESTAMPDIFF for PostgreSQL or Oracle found in Internet, it's sad but them do not work like in MySQL

**Comment by x86demon** [comment]
Fixed in #39
